packages/components/src/components/domainproperties/dataset/DatasetDesignerPanels.tsx

```diff
@@ -55,13 +73,136 @@ export class DatasetDesignerPanelImpl extends React.PureComponent<Props & Inject
+
+        importData({
+            schemaName: 'study',
+            queryName: model.name,
+            file: fileImportData,
+            importUrl: ActionURL.buildURL('study', 'import', getServerContext().container.path, { name: model.name }),
+        })
+            .then(response => {
+                setSubmitting(false, () => {
+                    this.props.onComplete(model);
+                });
+            })
+            .catch(error => {
+                console.error(error);
+                setSubmitting(false, () => {
+                    this.props.onComplete(model, resolveErrorMessage(error));
+                });
+            });
+    }
+
+    saveDomain = () => {
+        const { setSubmitting } = this.props;
+        const { model, fileImportData } = this.state;
+
+        saveDomain(model.domain, model.getDomainKind(), model.getOptions(), model.name)
+            .then(response => {
+                let updatedModel = model.set('exception', undefined) as DatasetModel;
+                updatedModel = updatedModel.merge({ domain: response }) as DatasetModel;
+                this.setState(() => ({ model: updatedModel }));
+
+                // If we're importing List file data, import file contents
+                if (fileImportData) {
+                    this.handleFileImport();
+                } else {
+                    setSubmitting(false, () => {
+                        this.props.onComplete(updatedModel);
+                    });
+                }
+            })
+            .catch(response => {
+                const exception = resolveErrorMessage(response);
+                const updatedModel = exception
+                    ? (model.set('exception', exception) as DatasetModel)
+                    : (model.merge({ domain: response, exception: undefined }) as DatasetModel);
+
+                setSubmitting(false, () => {
+                    this.setState(() => ({ model: updatedModel }));
+                });
+            });
+    };
+
+    render() {
+        const {
+            useTheme,
+            onTogglePanel,
+            visitedPanels,
+            submitting,
+            onCancel,
+            currentPanelIndex,
+            firstState,
+            validatePanel,
+            containerTop,
+            successBsStyle,
+        } = this.props;
+
+        const { model } = this.state;
+
+        return (
+            <BaseDomainDesigner
+                name={model.name}
+                exception={model.exception}
+                domains={List.of(model.domain)}
+                hasValidProperties={model.hasValidProperties()}
+                visitedPanels={visitedPanels}
+                submitting={submitting}
+                onCancel={onCancel}
+                onFinish={this.onFinish}
+                saveBtnText="Save"
+                successBsStyle={successBsStyle}
+            >
                 <DatasetPropertiesPanel
                     initCollapsed={false}
```

Comment:
```suggestion
                    initCollapsed={currentPanelIndex !== 0}
```

Reply:
Also, for the `panelStatus` prop, see the other designers, we'll want to make this one match those (i.e. see usages of `getDomainPanelStatus`).

Reply:
Same for the `validate` property, we'll want to match the other designers.

Reply:
Also add `successBsStyle={successBsStyle}`